R/create_query_data.R

```diff
@@ -0,0 +1,894 @@
+      )
+    abort(msg)
+  }
+}
+
+#' Create an `query` object
+#'
+#' A `query` object defines a query, e.g., a Standard MedDRA Query (SMQ), a
+#' Standardised Drug Grouping (SDG), or a customized query (CQ). It can be used
+#' as input to `create_query_data()`.
+#'
+#' @param prefix Prefix of the new variables, e.g., `"SMQ03"`
+#'
+#' @param name The variable `<prefix>NAM` is set to the specified value for
+#'   observations matching the query definition. If the `auto` keyword is
+#'   specified, the variable is set to the name of the query in the SMQ/SDG
+#'   database.
+#'
+#'   *Permitted Values*: A character scalar or the `auto` keyword. The `auto`
+#'   keyword is permitted only for queries which are defined by a `smq_select()`
+#'   or `sdg_select()` object.
+#'
+#' @param id The variable `<prefix>CD` is set to the specified value for
+#'   observations matching the query definition if the parameter is specified.
+#'   Otherwise the variable is not created. If the `auto` keyword is specified,
+#'   the variable is set to the id of the query in the SMQ/SDG database.
+#'
+#'   *Permitted Values*: A integer scalar or the `auto` keyword. The `auto`
+#'   keyword is permitted only for queries which are defined by a `smq_select()`
+#'   or `sdg_select()` object.
+#'
+#' @param add_scope_num The variable `<prefix>SCN` is added to the output
+#'   dataset if the parameter is specified. Otherwise the variable is not
+#'   created.
+#'
+#'   The value is set to `1` if the scope is broad. Otherwise, it is set to `2`.
+#'
+#'   If the parameter is set to `TRUE`, the definition must be a `smq_select()`
+#'   object.
+#'
+#'   *Default*: `FALSE`
+#'
+#'   *Permitted Values*: `TRUE`, `FALSE`
+#'
+#' @param definition Definition of terms belonging to the query
+#'
+#'   There are four different ways to define the terms:
+#'
+#'   * A `smq_select` object is specified to select a query from the SMQ
+#'     database. The database contains Standard MedDRA Queries (SMQs) and AE
+#'     Grouping Terms (AEGTs).
+#'
+#'   * A `sdg_select` object is specified to select a query from the SDG
+#'     database.
+#'
+#'   * A data frame with columns `TERM_LEVEL` and `TERM_NAME` or `TERM_ID` can
+#'     be specified to define the terms of the query. The `TERM_LEVEL` should be
+#'     set to the name of the variable which should be used to select the terms,
+#'     e.g., `"AEDECOD"` or `"AELLTCD"`. `TERM_LEVEL` does not need to be
+#'     constant within a query. For example a query can be based on `AEDECOD`
+#'     and `AELLT`.
+#'
+#'     If `TERM_LEVEL` refers to a character variable, `TERM_NAME` should be set
+#'     to the value the variable. If it refers to a numeric variable, `TERM_ID`
+#'     should be set to the value of the variable. If only character variables
+#'     or only numeric variables are used, `TERM_ID` or `TERM_NAME` respectively
+#'     can be omitted.
+#'
+#'   * A list of data frames and `smq_select` objects can be specified to define
```

Comment:
so we dont expect a list of dataframes + sdg?

Reply:
At the moment we do not expect it. Are you aware of any use cases?